binary/parameterized_types.proto

```diff
@@ -0,0 +1,117 @@
+syntax = "proto3";
+package io.substrait;
+
+import "type.proto";
+import "extensions.proto";
+
+option java_multiple_files = true;
+
+message ParameterizedType {
```

Comment:
Is a type also a parameterized type?  In other words, in spots expecting a `ParameterizedType` can I just give a `Type`?  Is `VARCHAR(20)` a `ParameterizedType`?

Reply:
From a proto idl, you can represent VARCHAR(20) three different ways: with Type, ParameterizedType and DerivationExpression. They are very similar representations but to ensure that only the right kind of type information is input in each location, they are independent in the idl. I'm not sure of a better way to do this. The alternative is merging them all and having to write separate tools to validate semantics.